cmd/kubeadm/app/util/version.go

```diff
@@ -61,7 +100,7 @@ var (
 //  latest      (latest release, including alpha/beta)
 //  latest-1    (latest release in 1.x, including alpha/beta)
 //  latest-1.0  (and similarly 1.1, 1.2, 1.3, ...)
-func KubernetesReleaseVersion(version string) (string, error) {
+func KubernetesReleaseVersion(version string, fetcher URLFetcher) (string, error) {
```

Comment:
The whole idea of the interface seems like an overkill to me. You need just a single function. It's unlikely (at least for me) that `KubernetesReleaseVersion` will require something different for mocking. Also, ideally we don't want users of `KubernetesReleaseVersion` to construct and pass custom fetchers.

What I propose is to move the entire body of `KubernetesReleaseVersion` to a package local implementation function that takes fetcher func as an additional parameter. Then simply implement `KubernetesReleaseVersion` as a one liner that calls the internal implementation with `fetchFromURL`.

Thus the test cases can now use the internal implementation with mocked URL fetcher and provide tests for the edge cases.
Here is something similar that I did a while ago:
https://github.com/kubernetes/kubernetes/blob/46a80259f681736b4669efa49304bcb4218bab34/cmd/kubeadm/app/util/runtime/runtime.go#L235